activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -150,7 +150,7 @@ def locking_column=(value)
 
         # The version column used for optimistic locking. Defaults to +lock_version+.
         def locking_column
-          reset_locking_column unless defined?(@locking_column)
+          @locking_column = DEFAULT_LOCKING_COLUMN unless defined?(@locking_column)
```

Comment:
@rafaelfranca I am not sure if you asked do I know why that is wanted, or literally how is that done.

- `locking_column` calls `reset_locking_column`
- `reset_locking_column` calls `locking_column=`
- `locking_column=` calls `reload_schema_from_cache`

I'm looking into writing tests for this. but they are a little tricky